python/gtsam/tests/test_DiscreteConditional.py

```diff
@@ -36,6 +43,53 @@ def test_single_value_versions(self):
         actual = conditional.sample(2)
         self.assertIsInstance(actual, int)
 
+    def test_multiply(self):
+        """Check calculation of joint P(A,B)"""
+        conditional = DiscreteConditional(A, [B], "1/2 2/1")
+        prior = DiscreteConditional(B, "1/2")
+
+        # P(A,B) = P(A|B) * P(B) = P(B) * P(A|B)
+        for actual in [prior * conditional, conditional * prior]:
+            self.assertEqual(2, actual.nrFrontals())
+            for v, value in actual.enumerate():
+                self.assertAlmostEqual(actual(v), conditional(v) * prior(v))
+
+    def test_multiply2(self):
+        """Check calculation of conditional joint P(A,B|C)"""
+        A_given_B = DiscreteConditional(A, [B], "1/3 3/1")
+        B_given_C = DiscreteConditional(B, [C], "1/3 3/1")
+
+        # P(A,B|C) = P(A|B)P(B|C) = P(B|C)P(A|B)
+        for actual in [A_given_B * B_given_C, B_given_C * A_given_B]:
+            self.assertEqual(2, actual.nrFrontals())
+            self.assertEqual(1, actual.nrParents())
+            for v, value in actual.enumerate():
+                self.assertAlmostEqual(actual(v), A_given_B(v) * B_given_C(v))
+
+    def test_multiply4(self):
+        """Check calculation of joint P(A,B,C|D,E) = P(A,B|D) P(C|D,E)"""
+        A_given_B = DiscreteConditional(A, [B], "1/3 3/1")
+        B_given_D = DiscreteConditional(B, [D], "1/3 3/1")
+        AB_given_D = A_given_B * B_given_D
+        C_given_DE = DiscreteConditional(C, [D,  E], "4/1 1/1 1/1 1/4")
+
+        # P(A,B,C|D,E) = P(A,B|D) P(C|D,E) = P(C|D,E) P(A,B|D)
+        for actual in [AB_given_D * C_given_DE, C_given_DE * AB_given_D]:
+            self.assertEqual(3, actual.nrFrontals())
+            self.assertEqual(2, actual.nrParents())
+            for v, value in actual.enumerate():
+                self.assertAlmostEqual(
+                    actual(v), AB_given_D(v) * C_given_DE(v))
+
+    def test_marginals(self):
+        conditional = DiscreteConditional(A, [B], "1/2 2/1")
+        prior = DiscreteConditional(B, "1/2")
+        pAB = prior * conditional
+        self.gtsamAssertEquals(prior, pAB.marginal(B[0]))
+
+        pA = DiscreteConditional(A % "5/4")
```

Comment:
What is happening over here? We have `(0, 2) % "5/4"`, not sure if this should be `(0, 2), "5/4"`.

Reply:
Yeah, will fix that

Reply:
I am making fixes since other stuff is also breaking. 🙂